test_mixpanel.py

```diff
@@ -449,37 +450,49 @@ class TestConsumer:
     def setup_class(cls):
         cls.consumer = mixpanel.Consumer(request_timeout=30)
 
-    @contextlib.contextmanager
-    def _assertSends(self, expect_url, expect_data, consumer=None):
-        if consumer is None:
-            consumer = self.consumer
-
-        mock_response = Mock()
-        mock_response.data = six.b('{"status": 1, "error": null}')
-        with patch('mixpanel.urllib3.PoolManager.request', return_value=mock_response) as req:
-            yield
-
-            assert req.call_count == 1
-            (call_args, kwargs) = req.call_args
-            (method, url) = call_args
-            assert method == 'POST'
-            assert url == expect_url
-            assert kwargs["fields"] == expect_data
-
+    @responses.activate
     def test_send_events(self):
-        with self._assertSends('https://api.mixpanel.com/track', {"ip": 0, "verbose": 1, "data": '{"foo":"bar"}'}):
-            self.consumer.send('events', '{"foo":"bar"}')
-
+        responses.add(
+            responses.POST,
+            'https://api.mixpanel.com/track',
+            json={"status": 1, "error": None},
+            status=200,
+            match=[responses.json_params_matcher({"ip": 0, "verbose": 1, "data": '{"foo":"bar"}'})],
+        )
+        self.consumer.send('events', '{"foo":"bar"}')
```

Comment:
I haven't used `responses` before and couldn't find it in the docs - looking at this test case, there isn't any obvious assertion that the registered response was actually called. If `self.consumer.send` doesn't make any network requests at all, will this test fail? Is that what the `activate` decorator is supposed to do?

(I do see at https://github.com/getsentry/responses#assertions-on-declared-responses "When used as a context manager, Responses will, by default, raise an assertion error if a url was registered but not accessed" but don't see anything in the readme about the decorator behavior)

Reply:
Oh yeah, you are correct. Thx.